packages/compass-data-modeling/src/services/data-model-storage.ts

```diff
@@ -1,4 +1,96 @@
 import { z } from '@mongodb-js/compass-user-data';
+import type { MongoDBJSONSchema } from 'mongodb-schema';
+
+export const RelationshipSideSchema = z.object({
+  ns: z.string(),
+  cardinality: z.number(),
+  fields: z.array(z.string()),
+});
+
+export type RelationshipSide = z.output<typeof RelationshipSideSchema>;
+
+export const RelationshipSchema = z.object({
+  id: z.string(),
+  relationship: z.tuple([RelationshipSideSchema, RelationshipSideSchema]),
+  isInferred: z.boolean(),
+});
+
+export type Relationship = z.output<typeof RelationshipSchema>;
+
+export const StaticModelSchema = z.object({
+  collections: z.array(
+    z.object({
+      ns: z.string(),
+      jsonSchema: z.unknown(), // skipped for simplicity
```

Comment:
I understand that we probably don't want a full blown validation of `MongoDBJSONSchema` type implemented here, but I think we might want to consider using `z.custom` here, provide the MongoDBJSONSchema as a generic argument and maybe adding some very basic validation for the type matching. The purpose of these `z.<type>` helpers is not only to get the type definition after all, but to actually validate that what, in our case, you're loading from disk (and at some point user will be passing to the app) is matching the shape that we need for this to function.